x/coin.go

```diff
@@ -86,11 +90,17 @@ func (c Coin) Multiply(times int64) Coin {
 		frac = frac % FracUnit
 	}
 
-	return Coin{
+	wantPositive := times > 0 == (c.Whole >= 0 && c.Fractional >= 0)
```

Comment:
Nice line... took me a second to decipher... You can also use `c.IsNonNegative()` for the second half of the compare... I think it reads a bit nicer.

Reply:
This code is gone :see_no_evil: